src/main/java/org/spongepowered/common/world/DimensionManager.java

```diff
@@ -24,336 +24,1002 @@
+            if (!Files.exists(worldFolder)) {
+                unregisterWorldProperties(this.props);
+                return true;
+            }
+
+            try {
+                FileUtils.deleteDirectory(worldFolder.toFile());
+                unregisterWorldProperties(this.props);
+                return true;
+            } catch (IOException e) {
+                return false;
             }
         }
 
-        return dim;
     }
 
-    public static void sendDimensionRegistration(WorldServer worldserver, EntityPlayerMP playerIn, int dim) {
-//        // register dimension on client-side
-//        FMLEmbeddedChannel serverChannel = NetworkRegistry.INSTANCE.getChannel("FORGE", Side.SERVER);
-//        serverChannel.attr(FMLOutboundHandler.FML_MESSAGETARGET).set(FMLOutboundHandler.OutboundTarget.PLAYER);
-//        serverChannel.attr(FMLOutboundHandler.FML_MESSAGETARGETARGS).set(playerIn);
-//        serverChannel.writeOutbound(new ForgeMessage.DimensionRegisterMessage(dimension,
-//                ((SpongeDimensionType) ((Dimension) worldserver.provider).getType()).getDimensionTypeId()));
+    public static DimensionType getClientDimensionType(DimensionType serverDimensionType) {
+        switch (serverDimensionType) {
+            case OVERWORLD:
+            case NETHER:
+            case THE_END:
+                return serverDimensionType;
+            default:
+                return DimensionType.OVERWORLD;
+        }
+    }
+
+    public static void sendDimensionRegistration(EntityPlayerMP playerMP, DimensionType dimensionType) {
+        // Do nothing in Common
+    }
+
+    public static void loadDimensionDataMap(@Nullable NBTTagCompound compound) {
+        dimensionBits.clear();
+        if (compound == null) {
+            for (int dimensionId : dimensionTypeByDimensionId.keys()) {
+                if (dimensionId >= 0) {
+                    dimensionBits.set(dimensionId);
+                }
+            }
+        } else {
+            final int[] intArray = compound.getIntArray("DimensionArray");
+            for (int i = 0; i < intArray.length; i++) {
+                for (int j = 0; j < Integer.SIZE; j++) {
+                    dimensionBits.set(i * Integer.SIZE + j, (intArray[i] & (1 << j)) != 0);
+                }
+            }
+        }
+    }
+
+    public static NBTTagCompound saveDimensionDataMap() {
+        int[] data = new int[(dimensionBits.length() + Integer.SIZE - 1 )/ Integer.SIZE];
+        NBTTagCompound dimMap = new NBTTagCompound();
+        for (int i = 0; i < data.length; i++) {
+            int val = 0;
+            for (int j = 0; j < Integer.SIZE; j++) {
+                val |= dimensionBits.get(i * Integer.SIZE + j) ? (1 << j) : 0;
+            }
+            data[i] = val;
+        }
+        dimMap.setIntArray("DimensionArray", data);
+        return dimMap;
+    }
+
+    public enum RegisterDimensionTypeResult {
```

Comment:
TODO for myself. Remove these result enums or figure out how this correlates to a WorldResult mechanism above.